### PR TITLE
chore: adapt to buildx rc1

### PR DESCRIPTION
**Pull request by jjmachan**
generated the docker file according to RC0 spec

### CODE REVIEW

File: bentoctl_sagemaker/create_deployable.py [hunk 1]
```diff
@@ -64,13 +40,40 @@ def create_deployable(
...
-    generate_sagemaker_service_in(deployable_path)
-    generate_serve_script_in(deployable_path)
+    shutil.copytree(bento_path, deployable_path, dirs_exist_ok=True)
+
+    bento_metafile = Path(bento_path, "bento.yaml")
+    with bento_metafile.open("r", encoding="utf-8") as metafile:
+        info = BentoInfo.from_yaml_file(metafile)
 
-    return docker_context_path
+    options = bentoml_cattr.unstructure(info.docker)
```
Comment (parano) [inline bentoctl_sagemaker/create_deployable.py:52]: Do we need to unstruct it here?

Comment (aarnphm) [reply bentoctl_sagemaker/create_deployable.py:52]: we do, because options is frozen

File: operator_config.py [hunk 2]
```diff
@@ -1,4 +1,53 @@
...
+    "timeout": {
+        "required": False,
+        "default": 60,
+        "coerce": int,
+        "type": "integer",
+        "help_message": "Timeout for the sagemaker deployment",
+    },
+}
+
+data_capture_schema = {
```
Comment (yubozhao) [inline operator_config.py:25]: This changes cause issue in `generate.py` line 41

File: bentoctl_sagemaker/create_deployable.py [hunk 3]
```diff
@@ -58,22 +37,43 @@ def create_deployable(
...
-    deployable_path = os.path.join(destination_dir, "bentoctl_deployable")
-    docker_context_path = deployable_path
+    deployable_path = Path(destination_dir)
 
     # copy over the bento bundle
-    shutil.copytree(bento_path, deployable_path)
-    dockerfile_path = generate_dockerfile_in(deployable_path, bento_metadata)
-    generate_sagemaker_service_in(deployable_path)
-    generate_serve_script_in(deployable_path)
+    shutil.copytree(bento_path, deployable_path, dirs_exist_ok=True)
```
Comment (yubozhao) [inline bentoctl_sagemaker/create_deployable.py:46]: This does not work for python 3.7

File: bentoctl_sagemaker/create_deployable.py [hunk 4]
```diff
@@ -58,22 +37,43 @@ def create_deployable(
...
+        dockerfile.write(
+            generate_dockerfile(
+                DockerOptions(**options).with_defaults(),
+                str(deployable_path),
+                use_conda=any(
+                    i is not None
+                    for i in bentoml_cattr.unstructure(info.conda).values()
+                ),
+            )
+        )
```
Comment (yubozhao) [inline bentoctl_sagemaker/create_deployable.py:66]: throw error:
```
Traceback (most recent call last):
  File "/usr/local/Caskroom/miniconda/base/envs/dev/bin/bentoctl", line 5, in <module>
    bentoctl()
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/utils.py", line 90, in wrapper
    return_value = func(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/utils.py", line 55, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/utils.py", line 24, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/cli/__init__.py", line 153, in build
    destination_dir=dist_dir,
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/deployment_config.py", line 218, in create_deployable
    overwrite_deployable=overwrite_deployable,
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoctl/operator/operator.py", line 128, in create_deployable
    bento_path, destination_dir, bento_metadata, overwrite_deployable
  File "/Users/bozhaoyu/src/aws-sagemaker-deploy/bentoctl_sagemaker/create_deployable.py", line 68, in create_deployable
    for i in bentoml_cattr.unstructure(info.conda).values()
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/bentoml/_internal/bento/gen.py", line 214, in generate_dockerfile
    return template.render(**get_templates_variables(options, use_conda=use_conda))
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/bozhaoyu/src/aws-sagemaker-deploy/bentoctl_sagemaker/sagemaker/template.j2", line 1, in top-level template code
    {% extends bento__dockerfile %}
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/loaders.py", line 195, in get_source
    pieces = split_template_path(template)
  File "/usr/local/Caskroom/miniconda/base/envs/dev/lib/python3.7/site-packages/jinja2/loaders.py", line 30, in split_template_path
    for piece in template.split("/"):
jinja2.exceptions.UndefinedError: 'bento__dockerfile' is undefined
```

Comment (aarnphm) [reply bentoctl_sagemaker/create_deployable.py:66]: we need an update on the extends field. I will fix that later td.

Comment (yubozhao) [reply bentoctl_sagemaker/create_deployable.py:66]: that fix is on sagemaker or on rc1?

Comment (yubozhao) [reply bentoctl_sagemaker/create_deployable.py:66]: sounds good. keep me posted. Feel free to tag along the fix for other two issues I mentioned.

File: bentoctl_sagemaker/create_deployable.py [hunk 5]
```diff
@@ -58,22 +43,40 @@ def create_deployable(
...
+
+    # copy sagemaker service.py
+    shutil.copy(
+        SERVICE_PATH,
+        os.path.join(deployable_path, "service.py"),
+    )
+
+    # then copy the serve script
+    shutil.copy(SERVE_PATH, dockerfile_path.parent.joinpath("serve"))
```
Comment (yubozhao) [inline bentoctl_sagemaker/create_deployable.py:78]: this is copying to a wrong location

File: bentoctl_sagemaker/create_deployable.py (hunk 3 above)
Comment (aarnphm) [reply bentoctl_sagemaker/create_deployable.py]: fixed with backports now

**Review by yubozhao — CHANGES_REQUESTED** [review]
Sagemaker continue throws `exec: "serve": executable file not found in $PATH`

Let's make sure the PATH is set correctly.

**Comment by jjmachan** [comment]
made a couple of changes and tested it on sagemaker. should be good to go

**Pull request merged**